lib/web/desktop.go

```diff
@@ -144,19 +145,29 @@ func proxyWebsocketConn(ws *websocket.Conn, con net.Conn) error {
 	// Ensure we send binary frames to the browser.
 	ws.PayloadType = websocket.BinaryFrame
 
+	var closeOnce sync.Once
+	close := func() {
+		ws.Close()
+		con.Close()
+	}
```

Comment:
why not defer the `Close()` calls here outside the goroutines? Alternatively, move them to the end of this function and check the `error` returns from `Close()`

Reply:
Because you actually get a different error (broken pipe) as opposed to a "the connection has been closed" error, which slips past our filtering.

Reply:
hmm right, and I suppose you may need to close within the goroutine to trigger an error on the other one and get it to stop